jax/_src/array.py

```diff
@@ -680,54 +680,85 @@ def make_array_from_callback(
 def make_array_from_single_device_arrays(
     shape: Shape, sharding: Sharding, arrays: Sequence[basearray.Array]
 ) -> ArrayImpl:
-  r"""Returns a ``jax.Array`` from a sequence of ``jax.Array``\s on a single device.
+  r"""Returns a ``jax.Array`` from a sequence of ``jax.Array``\s each on an individual device.
 
-  You can use this function if you have already ``jax.device_put`` the value on
-  a single device and want to create a global Array. The smaller ``jax.Array``\s should be
-  addressable and belong to the current process.
+  To use this function, every device in the global mesh must have an addressable array that
+  is already on on-device, typically either via ``jax.device_put`` or generated directly onto the device.
```

Comment:
I don't know what generated directly onto the device means.

I think it's sufficient to say device_put or "is already on the device"

Reply:
thanks!

Reply:
"typically either via ``jax.device_put``"